Ignite: org.apache.ignite.internal.binary.BinaryObjectImpl cannot be cast to java.lang.Comparable

```bash
Caused by: java.lang.ClassCastException: org.apache.ignite.internal.binary.BinaryObjectImpl cannot be cast to java.lang.Comparable
	at java.util.TreeMap.compare(TreeMap.java:1294)
	at java.util.TreeMap.put(TreeMap.java:538)
	at org.apache.hadoop.hdfs.db.ignite.RenameSubtreeINodes.apply(RenameSubtreeINodes.java:75)
	at org.apache.hadoop.hdfs.db.ignite.RenameSubtreeINodes.apply(RenameSubtreeINodes.java:24)
	at org.apache.ignite.internal.processors.closure.GridClosureProcessor$C1.execute(GridClosureProcessor.java:1756)
	at org.apache.ignite.internal.processors.job.GridJobWorker$2.call(GridJobWorker.java:598)
	at org.apache.ignite.internal.util.IgniteUtils.wrapThreadLoader(IgniteUtils.java:7085)
	at org.apache.ignite.internal.processors.job.GridJobWorker.execute0(GridJobWorker.java:592)
	... 12 more

	at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1516)
	at org.apache.hadoop.ipc.Client.call(Client.java:1462)
	at org.apache.hadoop.ipc.Client.call(Client.java:1372)
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:228)
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:116)
	at com.sun.proxy.$Proxy9.rename(Unknown Source)
	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.rename(ClientNamenodeProtocolTranslatorPB.java:592)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:422)
	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:165)
	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:157)
	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:95)
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:359)
	at com.sun.proxy.$Proxy10.rename(Unknown Source)
	at org.apache.hadoop.hdfs.server.namenode.NNThroughputBenchmark$RenameDirStats.executeOp(NNThroughputBenchmark.java:1176)
	at org.apache.hadoop.hdfs.server.namenode.NNThroughputBenchmark$StatsDaemon.benchmarkOne(NNThroughputBenchmark.java:461)
	at org.apache.hadoop.hdfs.server.namenode.NNThroughputBenchmark$StatsDaemon.run(NNThroughputBenchmark.java:444)
```

--- a/hadoop-hdfs-project/hadoop-hdfs-db/src/main/java/org/apache/hadoop/hdfs/db/ignite/BatchRenameINodes.java
+++ b/hadoop-hdfs-project/hadoop-hdfs-db/src/main/java/org/apache/hadoop/hdfs/db/ignite/BatchRenameINodes.java
@@ -27,7 +27,7 @@ public class BatchRenameINodes implements IgniteClosure<List<BinaryObject>, Stri
         BinaryObjectBuilder inodeKeyBuilder = ignite.binary().builder("InodeKey");
 
         Transaction tx = ignite.transactions().txStart(
-            TransactionConcurrency.OPTIMISTIC, TransactionIsolation.SERIALIZABLE);
+            TransactionConcurrency.PESSIMISTIC, TransactionIsolation.SERIALIZABLE);
 
         IgniteCache<BinaryObject, BinaryObject> inodesBinary = ignite.cache("inodes").withKeepBinary();
         for (int i = 0; i < inodes.size(); ++i) {

--- a/hadoop-hdfs-project/hadoop-hdfs-db/src/main/java/org/apache/hadoop/hdfs/db/ignite/RenameSubtreeINodes.java
+++ b/hadoop-hdfs-project/hadoop-hdfs-db/src/main/java/org/apache/hadoop/hdfs/db/ignite/RenameSubtreeINodes.java
@@ -1,8 +1,8 @@
 package org.apache.hadoop.hdfs.db.ignite;
 
 import java.util.List;
-import java.util.TreeMap;
-import java.util.TreeSet;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Set;
 import java.util.Map;
 import javax.cache.Cache;
@@ -31,7 +31,7 @@ public class RenameSubtreeINodes implements IgniteClosure<RenamePayload, String>
         IgniteCache<BinaryObject, BinaryObject> inodesBinary = ignite.cache("inodes").withKeepBinary();
         
         Transaction tx = ignite.transactions().txStart(
-            TransactionConcurrency.OPTIMISTIC, TransactionIsolation.SERIALIZABLE);
+            TransactionConcurrency.PESSIMISTIC, TransactionIsolation.SERIALIZABLE);
         
         // 1. query subtree inodes
         List<Cache.Entry<BinaryObject, BinaryObject>> result;
@@ -46,8 +46,8 @@ public class RenameSubtreeINodes implements IgniteClosure<RenamePayload, String>
         result = inodesBinary.query(scanAddress).getAll();
 
         // 2. update subtree inodes
-        Set<BinaryObject> keys = new TreeSet<>();
-        Map<BinaryObject, BinaryObject> map = new TreeMap<>();
+        Set<BinaryObject> keys = new HashSet<>();
+        Map<BinaryObject, BinaryObject> map = new HashMap<>();
         BinaryObjectBuilder inodeKeyBuilder = ignite.binary().builder("InodeKey");
         for (Cache.Entry<BinaryObject, BinaryObject> entry : result) {
             BinaryObject inodeValue = entry.getValue();

